src/components/implementation/no_interface/vkernel/micro_booter.c

```diff
@@ -17,25 +17,31 @@ int vmid;
 void
 vm_init(void *d)
 {
-	vmid = (int)d;
+	int         rcvd, blocked;
+	cycles_t    cycles;
+	thdid_t     tid;
+	tcap_time_t timeout = 0, thd_timeout;
+
+	vmid = cos_sinv(VM_CAPTBL_SELF_SINV_BASE, VK_SERV_VM_ID << 16 | cos_thdid(), 0, 0, 0);
```

Comment:
OK, but why not set a protocol of "first argument is option, second is thread id"?  Just want to save registers for actual use?